odoo/models.py

```diff
@@ -841,6 +841,27 @@ def _init_constraints_onchanges(cls):
         cls._ondelete_methods = BaseModel._ondelete_methods
         cls._onchange_methods = BaseModel._onchange_methods
 
+    @property
+    def _table_sql(self) -> SQL:
+        table_query = self._table_query
+        table_sql = SQL(f"({table_query})") if table_query else SQL.identifier(self._table)
+        if not self._depends:
+            return table_sql
+
+        to_flush_fields = []
```

Comment:
Comment: `add self._depends (and its transitive closure) as metadata to table_sql`